starfish/pipeline/target_assignment/point_in_poly.py

```diff
@@ -1,41 +1,73 @@
 import numpy as np
-from skimage.measure import points_in_poly
 import pandas as pd
+import regional
+from skimage.measure import points_in_poly
+from tqdm import tqdm
 
 from starfish.constants import Features
+from starfish.intensity_table import IntensityTable
 from ._base import TargetAssignmentAlgorithm
 
 
-class PointInPoly(TargetAssignmentAlgorithm):
-    def __init__(self, verbose=False, **kwargs):
-        self.verbose = verbose
+class PointInPoly2D(TargetAssignmentAlgorithm):
+
+    def __init__(self, **kwargs) -> None:
+        """
+        PointInPoly accepts no parameters, but all pipeline components must accept arbitrary kwargs
+        """
 
     @classmethod
-    def add_arguments(cls, parser):
+    def add_arguments(cls, parser) -> None:
         pass
 
     @staticmethod
-    def _assign(cells_region, spots, use_hull=True, verbose=False):
-        res = pd.DataFrame({'spot_id': range(0, spots.shape[0])})
-        res['cell_id'] = None
+    def _assign(
+            cells_region: regional.many, spots: pd.DataFrame, use_hull: bool=True, verbose:
+            bool=False
+    ) -> pd.DataFrame:
+
+        results = pd.DataFrame({'spot_id': range(0, spots.shape[0])})
+        results['cell_id'] = None
 
-        for cell_id in range(cells_region.count):
+        if verbose:
+            cell_iterator = tqdm(range(cells_region.count))
+        else:
+            cell_iterator = range(cells_region.count)
+
+        for cell_id in cell_iterator:
             if use_hull:
-                verts = cells_region[cell_id].hull
+                vertices = cells_region[cell_id].hull
             else:
-                verts = cells_region[cell_id].coordinates
-            verts = np.array(verts)
-            in_poly = points_in_poly(spots, verts)
-            res.loc[res.spot_id[in_poly], 'cell_id'] = cell_id
-            if verbose:
-                cnt = np.sum(in_poly)
-                print(cell_id, cnt)
+                vertices = cells_region[cell_id].coordinates
+            vertices = np.array(vertices)
+            in_poly = points_in_poly(spots, vertices)
+            results.loc[results.spot_id[in_poly], 'cell_id'] = cell_id
```

Comment:
If there are overlapping cells (which can definitely happen if we use a convex hull), this will assign a spot to the last cell it's in.  The current structure (spots gaining a unique `cell_id` column) will always have this problem, but an `assign_targets` dataframe could get around this if it's basically a list of spots that are associated with each cell.

Reply:
Nice catch. Referencing this for #374 .